core/conn/jdbc_type2/native/SrvrKds.cpp

```diff
@@ -193,3 +196,43 @@ void kdsCopyToSQLValueSeq(SQLValueList_def *SQLValueList,
 	SQLValueList->_length++;
 	FUNCTION_RETURN_VOID((NULL));
 }
+void kdsCopySQLErrorExceptionAndRowCount(
+							odbc_SQLSvc_SQLError *SQLError, 
+							char *msg_buf,
+							long sqlcode,
+							char *sqlState,
+							long currentRowCount)
+{
+	SRVRTRACE_ENTER(FILE_KDS+6);
+	ERROR_DESC_def *SqlErrorDesc;
+	size_t	len;
+
+	len = strlen(msg_buf);
+	// Allocate String Buffer
+	
+	SqlErrorDesc = (ERROR_DESC_def *)SQLError->errorList._buffer + SQLError->errorList._length;
+	SqlErrorDesc->errorText = new char[len+1];
+	if (SqlErrorDesc->errorText == NULL)
+	{
+		exit(0);
```

Comment:
Is this what the T2 driver code normally does on a memory allocation failure? How will we know that it was a memory allocation failure?